### PR TITLE
Removed inaccurate docstring from dispatcher.py and deprecated an unused...

### PR DESCRIPTION
**Pull request by apollo13**
... argument from disconnect.

### CODE REVIEW

**Comment by apollo13** [comment]
Buildbot, test this please.

**Comment by timgraham** [comment]
My opinion is that a deprecation period doesn't offer much benefit in a case like this as you can remove the `weak` parameter from an existing Django project without breaking backwards compatibility with older versions of Django.

File: django/dispatch/dispatcher.py [hunk 1]
```diff
@@ -149,12 +147,12 @@ def disconnect(self, receiver=None, sender=None, weak=True, dispatch_uid=None):
...
 
-            weak
-                The weakref state to disconnect
-
             dispatch_uid
                 the unique identifier of the receiver to disconnect
         """
+        if weak is not None:
+            warnings.warn("Passing `weak` to disconnect has no effect.",
+                RemovedInDjango20Warning)
```
Comment (berkerpeksag) [inline django/dispatch/dispatcher.py]: needs `stacklevel=2`

File: docs/releases/1.9.txt [hunk 2]
```diff
@@ -177,7 +177,7 @@ Miscellaneous
 Features deprecated in 1.9
 ==========================
 
-...
+* The ``weak`` argument to ``django.dispatch.signals.Sinal.disconnect`` has been deprecated.
```
Comment (claudep) [inline docs/releases/1.9.txt]: Sinal -> Signal

**Comment by claudep** [comment]
You implemented a special fast deprecation (deprecated in 1.9, removed in 2.0). If our dear release manager accepts it, we could then start the deprecation in 1.8 to have a "normal" deprecation period.

File: django/dispatch/dispatcher.py [hunk 3]
```diff
@@ -64,10 +66,6 @@ def connect(self, receiver, sender=None, weak=True, dispatch_uid=None):
 
                 Receivers must be able to accept keyword arguments.
 
-                If receivers have a dispatch_uid attribute, the receiver will
```
Comment (timgraham) [inline django/dispatch/dispatcher.py]: My understanding is that this comment refers to this code: https://github.com/django/django/blob/f8e4e4a935512aedf815f6aee85bc797b498ea75/django/dispatch/dispatcher.py#L130-L131

**Comment by timgraham** [comment]
Updated patch in https://github.com/django/django/pull/3981.

File: django/dispatch/dispatcher.py (hunk 3 above)
Comment (apollo13) [inline django/dispatch/dispatcher.py]: I don't think so and there is no code in there which try to access dispatch_uid on the receiver ever.

Comment (timgraham) [inline django/dispatch/dispatcher.py]: Okay, I see what you mean. I think the comment is meant to be "If receivers are connected with a dispatch_uid argument, the receiver will not be added if another receiver was already connected with that dispatch_uid." Does that make sense?

Comment (apollo13) [inline django/dispatch/dispatcher.py]: Ah, that explains my confusion -- Yeah your suggestion is good!

**Pull request closed without merging**